addons/stock/models/product_strategy.py

```diff
@@ -27,6 +27,10 @@ def _default_category_id(self):
     def _default_location_id(self):
         if self.env.context.get('active_model') == 'stock.location':
             return self.env.context.get('active_id')
+        if self.env['stock.warehouse'].search_count([]) == 1:
+            wh = self.env['stock.warehouse'].search([], limit=1)
```

Comment:
Take the warehouse of the company in `self.env.company` (defualt value of company_id field). it allows to have a correct default when several company are selected (if my first comment is valid).